x-pack/plugins/security/public/session/session_timeout.tsx

```diff
@@ -24,57 +28,168 @@ const GRACE_PERIOD_MS = 5 * 1000;
 const WARNING_MS = 60 * 1000;
 
 export interface ISessionTimeout {
-  extend(): void;
+  init(): void;
+  extend(url: string): void;
 }
 
 export class SessionTimeout {
-  private warningTimeoutMilliseconds?: number;
-  private expirationTimeoutMilliseconds?: number;
+  private channel: BroadcastChannel<SessionInfo>;
+  private elector: ReturnType<typeof LeaderElection.create>;
+  private sessionInfo?: SessionInfo;
+  private updateTimer?: number;
+  private warningTimer?: number;
+  private expirationTimer?: number;
   private warningToast?: Toast;
 
   constructor(
-    private sessionTimeoutMilliseconds: number | null,
     private notifications: NotificationsSetup,
     private sessionExpired: ISessionExpired,
     private http: HttpSetup
-  ) {}
+  ) {
+    const name = `${http.basePath.get()}/session_timeout`;
+    this.channel = new BroadcastChannel(name, { webWorkerSupport: false });
+    this.elector = LeaderElection.create(this.channel, {
+      fallbackInterval: 2000,
+      responseTime: 1000,
+    });
+  }
+
+  init() {
+    if (this.isAnon()) {
+      this.elector.die();
```

Comment:
That isn't strictly necessary, thought it may be better to explicitly close the elector when it's not being used.

However, I'm removing it to simplify things, seeing as the elector adds complexity and doesn't really give us a lot of benefit.